include/fastdds/rtps/common/WriteParams.h

```diff
@@ -19,119 +19,187 @@
-                        return *this;
-                    }
-
-                    const SampleIdentity& related_sample_identity() const
-                    {
-                        return related_sample_identity_;
-                    }
-
-                    SampleIdentity& related_sample_identity()
-                    {
-                        return related_sample_identity_;
-                    }
-
-                    static WriteParams WRITE_PARAM_DEFAULT;
-
-                private:
-
-                    SampleIdentity sample_identity_;
-
-                    SampleIdentity related_sample_identity_;
-            };
-
-        } //namespace rtps
-    } //namespace fastrtps
-} //namespace eprosima
+        related_sample_identity_ = std::move(sample_id);
+        return *this;
+    }
+
+    /**
+     * Get the value of the related_sample_identity member.
+     *
+     * @return Constant reference to the related_sample_identity member.
+     */
+    const SampleIdentity& related_sample_identity() const
+    {
+        return related_sample_identity_;
+    }
+
+    /**
+     * Set the value of the related_sample_identity member.
+     *
+     * @return Reference to the related_sample_identity member.
+     */
+    SampleIdentity& related_sample_identity()
+    {
+        return related_sample_identity_;
+    }
+
+    /**
+     * Get the value of the source_timestamp member.
+     *
+     * @return Current value of the source_timestamp member.
+     */
+    Time_t source_timestamp() const
+    {
+        return source_timestamp_;
+    }
+
+    /**
+     * Set the value of the source_timestamp member.
+     *
+     * @return Reference to the source_timestamp member.
+     */
+    Time_t& source_timestamp()
+    {
+        return source_timestamp_;
+    }
+
+    /**
+     * Set the source_timestamp member of this class.
+     *
+     * @param timestamp  New value for the source_timestamp member.
+     *
+     * @return Reference to the modified object in order to allow daisy chaining.
+     */
+    WriteParams& source_timestamp(
+            const Time_t& timestamp)
+    {
+        source_timestamp_ = timestamp;
+        return *this;
+    }
+
+    /**
+     * Set the source_timestamp member of this class.
+     *
+     * @param timestamp  New value for the source_timestamp member.
+     *
+     * @return Reference to the modified object in order to allow daisy chaining.
+     */
+    WriteParams& source_timestamp(
+            Time_t&& timestamp)
+    {
+        source_timestamp_ = std::move(timestamp);
+        return *this;
+    }
+
+    /**
+     * Default value for methods receiving a WriteParams.
+     *
+     * Will contain the following values on its members:
+     * - sample_identity: Invalid SampleIdentity
+     * - related_sample_identity: Invalid SampleIdentity
+     * - source_timestamp: Invalid Time_t
+     */
+    static WriteParams WRITE_PARAM_DEFAULT;
+
+private:
+
+    /// Holds sample_identity member value
```

Comment:
I think we should stick to the imperative or add `It`. English does not allow to omit the sentence's subject.